nexus/completion_test.go

```diff
@@ -92,6 +122,9 @@ func (h *failureExpectingCompletionHandler) CompleteOperation(ctx context.Contex
 	if completion.HTTPRequest.Header.Get("foo") != "bar" {
 		return HandlerErrorf(HandlerErrorTypeBadRequest, "invalid 'foo' header: %q", completion.HTTPRequest.Header.Get("foo"))
 	}
+	if completion.OperationID != "test-operation-id" {
+		return HandlerErrorf(HandlerErrorTypeBadRequest, "invalid %q header: %q", HeaderOperationID, completion.HTTPRequest.Header.Get(HeaderOperationID))
+	}
```

Comment:
Hmm.. I think I may have mislead you in the previous review. We should test that the links are delivered here, it's a different path in the caller for unsuccessful operations.